### PR TITLE
perfect getstarted and user-guides doc

### PR DESCRIPTION
**Pull request by Demontf**
This is my first contribution to this great open source project. I have done the following work. If have any problems, pls leave comments.
1. Fixed invalid links and incorrect sample code.
2. Standardize punctuation and smoothen the sentences.
3. Add tables to beautify the layout

### CODE REVIEW

**Comment by tianxiaoliang** [comment]
great, thanks

File: docs/getstarted/writing-rest.md [hunk 1]
```diff
@@ -100,7 +103,7 @@ servicecomb:
   service:
     name: RESTClient #name your consumer
 ```
-3.in main.go call your service
+3. in main.go call your service
```
Comment (tianxiaoliang) [inline docs/getstarted/writing-rest.md:106]: 当初不留空格是因为生成文档无论数字是什么，markdown都显示为1.
所以我就把空格都去掉了

File: docs/user-guides/service-discovery.md [hunk 2]
```diff
@@ -28,15 +28,15 @@ Service Discovery与Registry只能选择其一进行配置
 ```yaml
 servicecomb:
   registry:
-        type: pilotv2
-        address: grpc://istio-pilot.istio-system:15010
-        refeshInterval : 30s
+    type: pilotv2
```
Comment (tianxiaoliang) [inline docs/user-guides/service-discovery.md:31]: 这个其实没必要加空格或者制表符，下一个层级只需要2个空格

File: docs/user-guides/handler-chain.md [hunk 3]
```diff
@@ -21,26 +21,20 @@ Provider表示，当你被别人调用人，会通过的处理链
...
-router	路由策略
-
-loadbalance	负载均衡
-
-tracing-consumer	客户端调用链追踪
-
-transport	各协议客户端处理请求，如果你使用自定义处理链配置，那么结尾处必须加入这个handler
+|名称|功能|
+|:---:|:---:|
+|ratelimiter-consumer|客户端限流|
```
Comment (tianxiaoliang) [inline docs/user-guides/handler-chain.md:26]: sphinx文档框架不支持这个markdown的原生table写法

File: docs/getstarted/writing-rest.md (hunk 1 above)
Comment (Demontf) [reply docs/getstarted/writing-rest.md:106]: 是因为每个标题间的内容没有加缩进，所以md识别成重新开始计数了。那我去掉这类修改吧。

File: docs/getstarted/writing-rest.md [hunk 4]
```diff
@@ -65,9 +66,11 @@ servicecomb:
 6.In main.go init and start the chassis 
 ```go
 func main() {
+    // register struct
+    chassis.RegisterSchema("rest", &RestFulHello{})
     //start all server you register in server/schemas.
     if err := chassis.Init(); err != nil {
-        openlogging.Error("Init failed.", err)
+        openlogging.Error("Init failed. "+err.Error())
```
[the remark this answers is not in the document]
Comment (Demontf) [reply docs/getstarted/writing-rest.md:73]: 已统一成openlog

File: docs/user-guides/service-discovery.md (hunk 2 above)
Comment (Demontf) [reply docs/user-guides/service-discovery.md:31]: ok

**Review by tianxiaoliang — APPROVED** [review]

**Pull request merged**